File: shared/src/test/scala/test/oml/tables/BinaryScalarRestrictionAxiomTest.scala
/*
 * Copyright 2016 California Institute of Technology ("Caltech").
 * U.S. Government sponsorship acknowledged.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * License Terms
 */

package test.oml.tables

/*
 * Copyright 2016 California Institute of Technology ("Caltech").
 * U.S. Government sponsorship acknowledged.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * License Terms
 */

import gov.nasa.jpl.imce.oml.covariantTag
import gov.nasa.jpl.imce.oml.tables.{taggedTypes,BinaryScalarRestriction,BinaryScalarRestrictionHelper}
import scala.Predef.String
import org.scalacheck.Prop.forAll
import org.scalacheck.Properties

import scala.{None, Option, Some, StringContext}

object BinaryScalarRestrictionAxiomTest extends Properties("BinaryScalarRestriction") {

  property("construction") = forAll(
    SchemaGenerators.uuid,
    SchemaGenerators.uuid,
    SchemaGenerators.optInt_0_to_3,
    SchemaGenerators.optInt_0_to_3,
    SchemaGenerators.optInt_0_to_3,
    SchemaGenerators.uuid,
    SchemaGenerators.name)(
    (tboxUUID: java.util.UUID,
     uuid: java.util.UUID,
     length: Option[String],
     maxLength: Option[String],
     minLength: Option[String],
     restrictedScalarUUID: java.util.UUID,
     scalarName: String) => {
      val w = new BinaryScalarRestriction(
        covariantTag[taggedTypes.BinaryScalarRestrictionTag][String](uuid.toString),
        covariantTag[taggedTypes.TerminologyBoxTag][String](tboxUUID.toString),
        covariantTag[taggedTypes.DataRangeTag][String](restrictedScalarUUID.toString),
        length.map(covariantTag[taggedTypes.PositiveIntegerLiteralTag][String]),
        minLength.map(covariantTag[taggedTypes.PositiveIntegerLiteralTag][String]),
        maxLength.map(covariantTag[taggedTypes.PositiveIntegerLiteralTag][String]),
        covariantTag[taggedTypes.LocalNameTag][String](scalarName))
      val s = BinaryScalarRestrictionHelper.toJSON(w)
      val l = w.length match {
        case None => "null"
        case Some(n) => s"""\"$n\""""
      }
      val min = w.minLength match {
        case None => "null"
        case Some(n) => s"""\"$n\""""
      }
      val max = w.maxLength match {
        case None => "null"
        case Some(n) => s"""\"$n\""""
      }
      val t = s"""{"uuid":"${w.uuid}","tboxUUID":"${w.tboxUUID}","restrictedRangeUUID":"${w.restrictedRangeUUID}","length":$l,"minLength":$min,"maxLength":$max,"name":"${w.name}"}"""
      val r = BinaryScalarRestrictionHelper.fromJSON(s)
      (s == t) &&
        covariantTag.compareTaggedValues(w.tboxUUID, r.tboxUUID) &&
        (w.uuid == r.uuid) &&
        (w.length == r.length) &&
        (w.maxLength == r.maxLength) &&
        (w.minLength == r.minLength) &&
        covariantTag.compareTaggedValues(w.restrictedRangeUUID, r.restrictedRangeUUID) &&
        (w.name == r.name)
    })

}

File: js/test/js/BinaryScalarRestrictionAxiomTest.js
#!/usr/bin/env node
'use strict';

const OMF = require('../../../gov-nasa-jpl-imce-oml-tables').__ScalaJSExportsNamespace;

console.log("OMF Schema Tables...");

const w1 = BinaryScalarRestrictionAxiomJS.jsBinaryScalarRestrictionAxiom(
    "BCDEF-12345-6789A-012345",
    "12345-6789A-BCDEF-012345",
    0,
    10,
    2,
    "4567-2345-ABCD-1245",
    "1245-ABCD-2345-4567");

const s1 = OMF.gov.nasa.jpl.imce.omf.schema.tables.BinaryScalarRestrictionAxiomHelper().toJSON(w1);
const r1 = OMF.gov.nasa.jpl.imce.omf.schema.tables.BinaryScalarRestrictionAxiomHelper().fromJSON(s1);

console.log("W1: ", w1);
console.log("S1: ", s1);

console.log("Checking graphUUID...", w1.graphUUID === r1.graphUUID);
console.log("Checking uuid...", w1.uuid === r1.uuid);
console.log("Checking length...", w1.length === r1.length);
console.log("Checking maxLength...", w1.maxLength === r1.maxLength);
console.log("Checking minLength...", w1.minLength === r1.minLength);
console.log("Checking restrictedScalarUUID...", w1.restrictedScalarUUID === r1.restrictedScalarUUID);
console.log("Checking scalarUUID...", w1.scalarUUID === r1.scalarUUID);
